core/string/ustring.cpp

```diff
@@ -4322,15 +4316,15 @@ String String::pad_zeros(int p_digits) const {
 String String::trim_prefix(const String &p_prefix) const {
 	String s = *this;
 	if (s.begins_with(p_prefix)) {
-		return s.substr(p_prefix.length(), s.length() - p_prefix.length());
+		return s.right(-p_prefix.length());
 	}
 	return s;
 }
 
 String String::trim_suffix(const String &p_suffix) const {
 	String s = *this;
 	if (s.ends_with(p_suffix)) {
-		return s.substr(0, s.length() - p_suffix.length());
+		return s.left(-p_suffix.length());
```

Comment:
p_suffix is not longer than s, since it has to be a substring, meaning that the expression in substr could've evaluated to at most 0. The deal is similar with trim_prefix().